src/Lecture1.hs

```diff
@@ -94,7 +103,11 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str
+    | start < 0 = subString 0 end str
+    | end < 0   = []
+    | otherwise = reverse (take ((end - start) + 1) (reverse (take (end+1) str)))
```

Comment:
Reverse is a slow function 🐌 
You can write the function in a more performant way by using `drop`